packages/data/lib/src/repositories/movie_local_cache_repository_impl.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:data/src/database/dao/movie_dao.dart';
+import 'package:data/src/database/dto/movie_anticipated_dto.dart';
+import 'package:data/src/database/dto/movie_trending_dto.dart';
+import 'package:domain/domain.dart';
+
+class MovieLocalCacheRepositoryImpl implements MovieLocalCacheRepository {
+  final MovieDao movieDao;
+
+  const MovieLocalCacheRepositoryImpl({required this.movieDao});
+
+  @override
+  Future<List<MovieAnticipated>> getAnticipated() async {
+    final List<MovieAnticipatedDto> cachedList =
+        await movieDao.getAnticipatedList();
+    return cachedList.map((e) => e.toMovieAnticipated()).toList();
+  }
+
+  @override
+  Future<List<MovieTrending>> getTrending() async {
+    final List<MovieTrendingDto> cachedList = await movieDao.getTrendingList();
+    return cachedList.map((e) => e.toMovieTrending()).toList();
+  }
+
+  @override
+  Future<void> saveAnticipatedIntoCache(
+      List<MovieAnticipated> anticipated) async {
+    await movieDao.clearAnticipated();
```

Comment:
u shouldn't drop all the records. u should only drop the records which are not actual. and save new added records